### PR TITLE
Deduping concurrent write requests.

### LINKED ISSUE #43: Don't upload previously uploaded blobs
This should be configurable with a parameter, but the default behavior should be to avoid querying GetMissingBlobs for previously uploaded blobs, drastically reducing its time. However, GetMissingBlobs doesn't take all that long in itself, so this is lower priority. It is, however, very easy to implement.

### PR DESCRIPTION
**Pull request by ola-rozenfeld**
Fixes #43 and #44, because this also doesn't query GetMissingBlobs for
previously uploaded blobs.

### CODE REVIEW

File: go/pkg/client/cas_test.go [hunk 1]
```diff
@@ -375,6 +376,88 @@ func TestMissingBlobs(t *testing.T) {
...
+				go func() {
+					var input []*chunker.Chunker
+					for _, blob := range blobs {
+						input = append(input, chunker.NewFromBlob(blob, 10))
+					}
+					_, err := c.UploadIfMissing(ctx, input...)
+					if err != nil {
+						t.Errorf("c.UploadIfMissing(ctx, input) gave error %v, expected nil", err)
+					}
+					defer wg.Done()
```
Comment (ramymedhat) [inline go/pkg/client/cas_test.go:443]: defer at the top?

File: go/pkg/client/cas.go [hunk 2]
```diff
@@ -99,9 +168,27 @@ func (c *Client) UploadIfMissing(ctx context.Context, data ...*chunker.Chunker)
...
+		}
+		if v, _ := c.casUploadErrors.Load(dg); v != nil {
+			e, ok := v.(error)
+			if !ok {
+				err = fmt.Errorf("unexpected type found in error map")
+			}
+			err = e
+		}
+	}
+	log.V(2).Info("Done")
```
Comment (ramymedhat) [inline go/pkg/client/cas.go:191]: Can you make this more descriptive about what was finished?

**Review by ramymedhat — COMMENTED** [review]
Thanks! Did you run the tests with feature=race?

File: go/pkg/client/cas.go (hunk 2 above)
Comment (ola-rozenfeld) [reply go/pkg/client/cas.go]: Yep!

File: go/pkg/client/cas_test.go (hunk 1 above)
Comment (ola-rozenfeld) [reply go/pkg/client/cas_test.go]: Oh, right :-)

**Comment by ola-rozenfeld** [comment]
Yes, features=race and --runs_per_test=50.

**Review by ramymedhat — APPROVED** [review]

**Pull request merged**